File: dotnet/src/Tyouqu.Database.Abstractions/TyouquDatabaseException.cs
namespace Tyouqu.Database.Abstractions;

public class TyouquDatabaseException : Exception
{
    public TyouquDatabaseException(string message)
        : base(message)
    {
    }

    public TyouquDatabaseException(string message, Exception innerException)
        : base(message, innerException)
    {
    }

    public string? SqlId { get; set; }

    public string? Provider { get; set; }
}

File: java/src/main/java/com/tyouqu/database/TyouquDatabaseException.java
package com.tyouqu.database;

public class TyouquDatabaseException extends RuntimeException {
    private String sqlId;
    private String provider;

    public TyouquDatabaseException(String message) {
        super(message);
    }

    public TyouquDatabaseException(String message, Throwable cause) {
        super(message, cause);
    }

    public String sqlId() {
        return sqlId;
    }

    public TyouquDatabaseException sqlId(String sqlId) {
        this.sqlId = sqlId;
        return this;
    }

    public String provider() {
        return provider;
    }

    public TyouquDatabaseException provider(String provider) {
        this.provider = provider;
        return this;
    }
}
